<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="A programmatic interface to the Web Service methods provided by
    Bold Systems (&lt;http://www.boldsystems.org/&gt;) for genetic barcode data.
    Functions include methods for searching by sequences by taxonomic names,
    ids, collectors, and institutions; as well as a function for searching
    for specimens, and downloading trace files.">
<title>Interface to Bold Systems API • bold</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Interface to Bold Systems API">
<meta property="og:description" content="A programmatic interface to the Web Service methods provided by
    Bold Systems (&lt;http://www.boldsystems.org/&gt;) for genetic barcode data.
    Functions include methods for searching by sequences by taxonomic names,
    ids, collectors, and institutions; as well as a function for searching
    for specimens, and downloading trace files.">
<meta property="og:image" content="https://docs.ropensci.org/bold/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="index.html">bold</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.3.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="articles/bold.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/bold/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="bold">bold<a class="anchor" aria-label="anchor" href="#bold"></a>
</h1>
</div>
<p><a href="https://www.repostatus.org/#active" class="external-link"><img src="https://www.repostatus.org/badges/latest/active.svg" alt="Project Status: Active – The project has reached a stable, usable state and is being actively developed."></a> <a href="https://cran.r-project.org/package=bold" class="external-link"><img src="https://www.r-pkg.org/badges/version/bold" alt="cran version"></a> <a href="https://badges.cranchecks.info" class="external-link"><img src="https://badges.cranchecks.info/worst/bold.svg" alt="cran checks"></a> <a href="https://github.com/ropensci/bold/actions/" class="external-link"><img src="https://github.com/ropensci/bold/workflows/R-check/badge.svg" alt="R-check"></a> <a href="https://app.codecov.io/gh/ropensci/bold?branch=master" class="external-link"><img src="https://codecov.io/github/ropensci/bold/coverage.svg?branch=master" alt="codecov.io"></a> <a href="https://github.com/r-hub/cranlogs.app" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/bold" alt="rstudio mirror downloads"></a></p>
<p><code>bold</code> accesses BOLD barcode data.</p>
<p>The Barcode of Life Data Systems (BOLD) is designed to support the generation and application of DNA barcode data. The platform consists of four main modules: a data portal, a database of barcode clusters, an educational portal, and a data collection workbench.</p>
<p>This package retrieves data from the BOLD database of barcode clusters, and allows for searching of over 1.7M public records using multiple search criteria including sequence data, specimen data, specimen <em>plus</em> sequence data, as well as trace files.</p>
<p>Documentation for the BOLD API: <a href="http://v4.boldsystems.org/index.php/api_home" class="external-link uri">http://v4.boldsystems.org/index.php/api_home</a></p>
<p>See also the taxize book for more options for taxonomic workflows with BOLD: <a href="https://taxize.dev/" class="external-link uri">https://taxize.dev/</a></p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><strong>Installation instructions</strong></p>
<p><strong>Stable Version</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"bold"</span><span class="op">)</span></span></code></pre></div>
<p><strong>Development Version</strong></p>
<p>Install <code>sangerseqR</code> first if you’re planning on downloading and reading trace files. (The package is only used in the function <code><a href="reference/bold_trace.html">bold::bold_read_trace()</a></code>)</p>
<p>For R &lt; 3.5</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/source.html" class="external-link">source</a></span><span class="op">(</span><span class="st">"http://bioconductor.org/biocLite.R"</span><span class="op">)</span></span>
<span><span class="fu">biocLite</span><span class="op">(</span><span class="st">"sangerseqR"</span><span class="op">)</span></span></code></pre></div>
<p>For R &gt;= 3.5</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"sangerseqR"</span><span class="op">)</span></span></code></pre></div>
<p>Then install <code>bold</code></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ropensci/bold"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://docs.ropensci.org/bold/">"bold"</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="search-for-sequence-data-only">Search for sequence data only<a class="anchor" aria-label="anchor" href="#search-for-sequence-data-only"></a>
</h3>
<p>Default is to get a list back</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/bold_seq.html">bold_seq</a></span><span class="op">(</span>taxon<span class="op">=</span><span class="st">'Coelioxys'</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; $id</span></span>
<span><span class="co">#&gt; [1] "ABEE117-17"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $name</span></span>
<span><span class="co">#&gt; [1] "Coelioxys elongata"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $gene</span></span>
<span><span class="co">#&gt; [1] "ABEE117-17"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $sequence</span></span>
<span><span class="co">#&gt; [1] "------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------TTATCATTATATACATATCATCCTTCCCCATCAGTTGATTTAGCAATTTTTTYTTTACATTTATCAGGAATTTYTTYTATTATCGGATCAATAAATTTTATTGTAACAATTTTAATAATAAAAAATTATTCAATAAATTATAATCAAATACCTTTATTTCCATGATCAATTTTAATTACTACAATTTTATTATTATTATCATTACCTGTATTAGCAGGAGCTATTACAATATTATTATTTGATCGTAATTTAAATTCATCATTTTTTGACCCAATAGGAGGAGGAGATCCTATTTTATATCAACATTTATTTTG------------------------------------"</span></span></code></pre></div>
<p>You can optionally get back the <code>crul</code> response object</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/bold_seq.html">bold_seq</a></span><span class="op">(</span>taxon<span class="op">=</span><span class="st">'Coelioxys'</span>, response<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">response_headers</span></span>
<span><span class="co">#&gt; $status</span></span>
<span><span class="co">#&gt; [1] "HTTP/2 200 "</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $server</span></span>
<span><span class="co">#&gt; [1] "nginx"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $date</span></span>
<span><span class="co">#&gt; [1] "Mon, 20 Apr 2020 16:11:50 GMT"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`content-type`</span></span>
<span><span class="co">#&gt; [1] "application/x-download"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`x-powered-by`</span></span>
<span><span class="co">#&gt; [1] "PHP/5.3.15"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`content-disposition`</span></span>
<span><span class="co">#&gt; [1] "attachment; filename=fasta.fas"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`x-frame-options`</span></span>
<span><span class="co">#&gt; [1] "SAMEORIGIN"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`x-content-type-options`</span></span>
<span><span class="co">#&gt; [1] "nosniff"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`x-xss-protection`</span></span>
<span><span class="co">#&gt; [1] "1; mode=block"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="search-for-specimen-data-only">Search for specimen data only<a class="anchor" aria-label="anchor" href="#search-for-specimen-data-only"></a>
</h3>
<p>By default you download <code>tsv</code> format data, which is given back to you as a <code>data.frame</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/bold_specimens.html">bold_specimens</a></span><span class="op">(</span>taxon<span class="op">=</span><span class="st">'Osmia'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;      processid   sampleid recordID catalognum   fieldnum</span></span>
<span><span class="co">#&gt; 1  BEECA373-06 05-NT-0373   514740            05-NT-0373</span></span>
<span><span class="co">#&gt; 2  BEECA607-06 05-NT-0607   516959            05-NT-0607</span></span>
<span><span class="co">#&gt; 3  BEECA963-07 01-OR-0790   554153            01-OR-0790</span></span>
<span><span class="co">#&gt; 4  BEECB358-07 04-WA-1076   596920 BBSL697174 04-WA-1076</span></span>
<span><span class="co">#&gt; 5  BEECB438-07 00-UT-1157   597000 BBSL432653 00-UT-1157</span></span>
<span><span class="co">#&gt; 6 BEECC1176-09 02-UT-2849  1060879 BBSL442586 02-UT-2849</span></span>
<span><span class="co">#&gt;                    institution_storing collection_code      bin_uri</span></span>
<span><span class="co">#&gt; 1   York University, Packer Collection              NA BOLD:AAI2013</span></span>
<span><span class="co">#&gt; 2   York University, Packer Collection              NA BOLD:AAC8510</span></span>
<span><span class="co">#&gt; 3   York University, Packer Collection              NA BOLD:ABZ3184</span></span>
<span><span class="co">#&gt; 4 Utah State University, Logan Bee Lab              NA BOLD:AAC5797</span></span>
<span><span class="co">#&gt; 5 Utah State University, Logan Bee Lab              NA BOLD:AAF2159</span></span>
<span><span class="co">#&gt; 6   York University, Packer Collection              NA BOLD:AAE5368</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="search-for-specimen-plus-sequence-data">Search for specimen plus sequence data<a class="anchor" aria-label="anchor" href="#search-for-specimen-plus-sequence-data"></a>
</h3>
<p>By default you download <code>tsv</code> format data, which is given back to you as a <code>data.frame</code>. You can get the sequences in a separated list with <code>sepfasta=TRUE</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/bold_seqspec.html">bold_seqspec</a></span><span class="op">(</span>taxon<span class="op">=</span><span class="st">'Osmia'</span>, sepfasta<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">fasta</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="co">#&gt; $`BEECA373-06`</span></span>
<span><span class="co">#&gt; [1] "-ATTTTATATATAATTTTTGCTATATGATCAGGTATAATCGGATCAGCAATAAGAATTATTATTCGTATAGAATTAAGAATTCCTGGTTCATGAATTTCAAATGATCAAACTTATAACTCTTTAGTAACTGCTCATGCTTTTTTAATAATTTTTTTCTTAGTTATACCTTTTTTAATTGGAGGATTTGGAAATTGATTAATTCCTTTAATATTAGGAATCCCGGATATAGCTTTCCCTCGAATAAATAATATTAGATTTTGACTTTTACCCCCTTCATTAATATTATTACTTTTAAGAAATTTTATAAATCCAAGACCAGGTACTGGATGAACTGTTTATCCTCCTCTTTCTTCTCATTTATTTCATTCTTCTCCTTCAGTTGATATAGCCATTTTTTCTTTACATATTTCCGGTTTATCTTCTATTATAGGTTCGTTAAATTTTATTGTTACAATTATTATAATAAAAAATATTTCTTTAAAACATATCCAATTACCTTTATTTCCATGATCTGTTTTTATTACTACTATCTTATTACTTTTTTCTTTACCTGTTTTAGCAGGAGCTATTACTATATTATTATTTGATCGAAATTTTAATACTTCATTTTTTGATCCTACAGGAGGTGGAGATCCAATCCTTTATCAACATTTATTT"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`BEECA607-06`</span></span>
<span><span class="co">#&gt; [1] "AATATTATATATAATTTTTGCTTTGTGATCTGGAATAATTGGTTCATCTATAAGAATTATTATTCGTATAGAATTAAGAATTCCTGGTTCATGAATTTCAAATGATCAAGTTTATAATTCATTAGTTACAGCTCATGCTTTTTTAATAATTTTTTTTTTAGTTATACCATTTATAATTGGAGGATTTGGAAATTGATTAGTTCCTTTAATATTAGGAATTCCTGATATAGCTTTTCCTCGAATAAATAATATTAGATTTTGATTATTACCACCATCATTAATACTTTTACTTTTAAGAAATTTTTTTAATCCAAGTTCAGGAACTGGATGAACTGTATATCCTCCTCTTTCATCATATTTATTTCATTCTTCACCTTCTGTTGATTTAGCTATTTTTTCTCTTCATATATCAGGTTTATCTTCTATTATAGGTTCATTAAACTTTATTGTAACTATTATTATAATAAAAAATATTTCTTTAAAGTATATTCAATTGCCATTATTTCCATGATCTGTTTTTATTACTACAATTCTTTTATTATTATCATTACCAGTTTTAGCAGGTGCTATTACTATATTATTATTTGATCGAAATTTTAATACTTCATTTTTTGATCCTACAGGAGGGGGAG--------------------------"</span></span></code></pre></div>
<p>Or you can index to a specific sequence like</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span><span class="op">$</span><span class="va">fasta</span><span class="op">[</span><span class="st">'GBAH0293-06'</span><span class="op">]</span></span>
<span><span class="co">#&gt; $`GBAH0293-06`</span></span>
<span><span class="co">#&gt; [1] "------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------TTAATGTTAGGGATTCCAGATATAGCTTTTCCACGAATAAATAATATTAGATTTTGACTGTTACCTCCATCTTTAATATTATTACTTTTAAGAAATTTTTTAAATCCAAGTCCTGGAACAGGATGAACAGTTTATCCTCCTTTATCATCAAATTTATTTCATTCTTCTCCTTCAGTTGATTTAGCAATTTTTTCTTTACATATTTCAGGTTTATCTTCTATTATAGGTTCATTAAATTTTATTGTTACAATTATTATAATAAAAAATATTTCTTTAAAATATATTCAATTACCTTTATTTTCTTGATCTGTATTTATTACTACTATTCTTTTATTATTTTCTTTACCTGTATTAGCTGGAGCTATTACTATATTATTATTTGATCGAAATTTTAATACATCTTTTTTTGATCCAACAGGAGGGGGAGATCCAATTCTTTATCAACATTTATTTTGATTTTTTGGTCATCCTGAAGTTTATATTTTAATTTTACCTGGATTTGGATTAATTTCTCAAATTATTTCTAATGAAAGAGGAAAAAAAGAAACTTTTGGAAATATTGGTATAATTTATGCTATATTAAGAATTGGACTTTTAGGTTTTATTGTT---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------"</span></span></code></pre></div>
<p>Note: you can get several sequences for one process id. For example:</p>
<pre class="{r}"><code><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/bold_seqspec.html">bold_seqspec</a></span><span class="op">(</span>taxon<span class="op">=</span><span class="st">'Salix'</span>, sepfasta<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">fasta</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">fasta</span><span class="op">)</span> <span class="op">==</span> <span class="st">"SALIX204-08"</span><span class="op">]</span></span>
<span><span class="co">#&gt; $SALIX204-08</span></span>
<span><span class="co">#&gt; CCGCCCCTAA------ATTTACAAATTACCAAGATTCGAATTTATAACATTTACTATTCAATTTAACC-------------------TTTTTTTATAAGTCTTA-------------------------TAAGAAAGAAATACAAAAGTAAAAAAAAAA-TTATTTAAGGAGCAATGCCAACCCTCTTGATAGAACAAGAAATTGGCGATTGCTCCTTTTTTTTTTTT--CAAAAGCTCGTACACACTAAGACAAAAGTCTTATCCATTTACAAAAGTCTTATCCATTTGT</span></span>
<span><span class="co">#&gt; $SALIX204-08</span></span>
<span><span class="co">#&gt; [1] ATTGGAACGCCAAGTTGCTCTTGATTCGGGGGTTCCTGCTATAGCCGAACACGAGGGAAAGATAATTTATACCGACATTGACAAGATTATTTTATCGGGGAATGGTTATACTGTAAGCATTCCATTAGTTATGTATCAACGTTCCAACAAAAATACCTGTATGCATCAAAAAACTCAGGTTCAGCGAGGAAAATGCATTAAAAGGGGACAAGTTTTGGCGGACGGTGCCGCTACAGTTGGTGGCGAACTCGCCTTGGGCAAAAACATATTAGTAGCTTATATGCCATGGGAAGGCTACAATTTTGAAGATGCGGTACTCATTAGCGAACGTCTGGTATATGAAGATGTTTATACTTCT</span></span>
<span><span class="co">#&gt; $SALIX204-08</span></span>
<span><span class="co">#&gt; [1] TATTGGATTTGTTGCTAAAATATCGGTATTAAACCCGAAACTCCCGGCGGATGACCAGTGACCCAAGGAAACGAAAGAATCGGTTATATTTTTCATACGATCTCCTCTTATTATAGACAGACTAATTATCTATTTATTTTTTTTTTT---TTAT-TATTCATTTACCTATTTCTAAATAGTAACTAGAGTCAAAAATATATTCAATTTCTATTAA--------------------TTAGAAATTTATTTTTTGTTCAATTGTAAATTTCTAATTAAGAACAATACTTAGTAAAATTGGGTATCAGGTCTTGTGCCAATTGCAAAATATCTACTTTA--TTTTTTTTTGTTACAACACTTCCTTAAAAAAGT-TTGGATTGGACTAACGAAGAAAGGGGAAGGAAGAAAGTGAATCAGTATACTAATTCCTCATCCCCCAATCAATCCTTCCCGTTGGGTTATTGTCCCAAT-AAAAATAAAACGAAATAAATAATTGTAGGAGTAAAATCTTGATAGAATTCGAAAAAGCAAGTACAAGAAATAAT-AAAAAAAAAATACGTA----------TTTTTTTTTTATTTTATAGGATTAGATTAAACAAAT</span></span>
<span><span class="co">#&gt; $SALIX204-08</span></span>
<span><span class="co">#&gt; [1] CTGCTGTAAGTTTTCGATGAAATTTTGAATACTATCCTAGAATAATTAATAATTCATGAGTTATTCAAGAGAAAAAATTCTACTAATTGATAAGATCAGATA--------------------AGTCTTCTAGTTCTTTCTAGTTCTTTATAGTCTAGGCCCTTGATTCAAACATTGAAGTTATTGTATAAACGTGAAAAATCTGGATTACCTACCCCATCTCCTCATTCTGAACTTTTTTCCATTCTATT----------------AAAAAAAACCTATTCGGTTAGATCCACCCGAAATAT------GGAATTTTCGGTATAAAAGCAAATTTTTGGCACACAAGACTCGACTCTTATTACA--------------------AATGAATTTAGAAAAATGCTTTTCTATTTCGAAAAAGTTCTAGAAACCACTTCATTTCTTGGTGTCAAAATGGGATATATGGTATAAATATAGAGAATCTATTTTCTTTTTTTCCAAACAAAAAAAAAA----GATCTTGGAGATTGTCTA</span></span>
<span><span class="co">#&gt; $SALIX204-08</span></span>
<span><span class="co">#&gt; [1] TGAGACTATGCTTGGCAAACGGGTTGATTATTCGGGGCGTTCCGTTATTGTCGTAGGCCCCTCACTTTCATTACATCGATGTGGATTGCCTCGCGAAATTGCAATAGAGCTTTTCCAGACATTTGTAATTCGTGGTCTAATTAGGCAACATCTTGCTTCAAACATAGGAGTTGCTAAGAGTAAAATTCGGGAAAAAGAGCCAATTGTATGGGGAATACTTCAGGAGGTTATGCGAGGACATCCTATATTGCTAAATAGAGCGCCTACTCTGCATAGATTAGGCATACAGGCATTCCAACCCATTTTAGTGGAAGGACGTGCTATTTGTTTACATCCATTAGTTCGTAAGGGATTCAATGCAGATTTTGATGGGGATCAAATGGCTGTTCATGTACCTTTATCGTTGGAGGCTCAAGCGGAGGCTCGTTTACTTATGTTTTCTCATATGAATCTCT</span></span>
<span><span class="co">#&gt; $SALIX204-08</span></span>
<span><span class="co">#&gt; [1] CAAACAGAAACTAAAGCAAGTGTTGGATTCAAGGCTGGTGTTAAAGATTATAAATTGACTTATTATACTCCTGAATATGAAACCAAAGATACTGATATCTTGGCAGCATTCCGAGTAACTCCTCAACCTGGAGTTCCGCCCGAGGAAGCAGGGGCCGCGGTAGCTGCTGAATCTTCTACTGGTACATGGACAACTGTGTGGACCGACGGGCTTACCAGTCTTGATCGTTATAAGGGACGATGCTACGACATCGAGCCCGTTGCTGGAGAAGAAAATCAATATATTGCTTATGTAGCTTACCCCTTAGACCTTTTTGAAGAAGGTTCTGTTACTAACATGTTTACTTCCATTGTGGGTAATGTATTTGGGTTCAAAGCCCTACGCGCTCTACGTCTAGAGGATTTGCGAATTCCTACTGCTTATGTTAAAACTTTTCAAGGCCCACCTCATGGTATCCAAGTTGAGAGAGATAAATTGAATAAGTATGGTCGCCCCCTATTGGGCTGTACTATTAAACCTAAATTGGGGTTATCCGCTAAGAATTACGGTAGAGCAGTTTATGAATGTCTACGCGGTGGACTTGATTTTACCAAAGATGATGAGAACGTGAACTCCCAACCATTTATGCGTTGGA</span></span>
<span><span class="co">#&gt; $SALIX204-08</span></span>
<span><span class="co">#&gt; [1] TTTCCACATTTAAATTATGTGTCAAATGGACTAATACCCTACCCCATCCATCTAGAAAAATTGGTTCAAATCCTTCGCTATTGGGTGAAAGATCCCTCCTCTTTGCATTTATTACGACTCTTTCTTCACGAGTATTGGAATTTGAACAGTCGTATTATTCCAAAGAAATCTATTTCTTTTTTTGCAAAAAAGACTCCAAGATTCTTCTTGTTCTTATATAATTCTCATGTATATGAATACGAGTCCGTTTTCTTTTTTCTTTGTAATCAATCCTTTCATTTCCGATTAACATTTTCTCAGGTCTTTCTTGAGCGAATATATTTCTATGGAAAAATAGAACATTTTGTA---------GAAGTCTTTACTAAGGATTGGGGGGACAGCCTATGCTTGCTCAAGGATCCTTTCATACATTATCTTAGATATCAAGGAAAATCCATTTTTGTCTCAAAGGATACGCCTCTTCTGATGAAAAAATGGAAATATTACCTTGTCAATTTATGTCAATGTCATTTTGATGTGTGCTTTCAACCCCCCAGGATCCATATAAACCCATTTTCATTATACAAGCATTCTTTCGCCTTATTAGGTTATCTTTCAAGTTCAAGTGTACGACTAAACCTTTCAGTGGTACGGAGTCAAATGCTAGAAAATGCATTTCTAATAGATAATATTATGAATAAACTCGATACAACAGTTTCAATTATTCCTTTGATTGGATCATTAGCAAAACTGAAATTTTGTAACGCAGTAGGACATCCCATTAGTAAACCGGCCTGGGCCGATTTTTCGGATTCTGATATTATCGACCGATTTGTCCGTATATGCAGAAATCTTTCTCATTATTATAGCGGATCCTCAAGAAAAAAGAGTTTGTATCGAATAAAATATATAC</span></span>
<span><span class="co">#&gt; $SALIX204-08</span></span>
<span><span class="co">#&gt; [1] TGCAGAATCCCGTGAACCATCGAGTCTTTGAACGCAAGTTGCGCCCGAGGCCTCCTGGTCGAGGGCACGTCTGCCT---GGGTGTCACGCATCGTCGCCCCC-------ACTCCCCTCG-------------------------GCTCACGAGGGCGGGGGCGGATACTGGTCTCCCGCGCGCTCCCGCCCGTGGTTGGCCT-AAAATCGAGTCCT-CGGCGGCGGTTGCC----ACGACAAGCGGTGGTTG-----------------------AGAGACCCTCGGACACGGTCGTGCGCGTGCTTG---------------------TCGCCCCCGGGAC--------------CTCCCGGACCCCCGAGCATTGGCTTTC-----------AAG-------------GATGCTCTCGTTGCGACCCCAGGTCAGGCGGGACTACCCGCTGAGTTTAA</span></span></code></pre>
<p>The reason being some process have different sequences for different markers.</p>
<pre class="{r}"><code><span><span class="va">res</span><span class="op">$</span><span class="va">data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">fasta</span><span class="op">)</span> <span class="op">==</span> <span class="st">"SALIX204-08"</span>, <span class="st">"markercode"</span><span class="op">]</span></span>
<span><span class="co">#&gt;[1] "trnH-psbA" "rpoB"      "atpF-atpH" "psbK-psbI" "rpoC1"     "rbcLa"     "matK"      "ITS2"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="get-trace-files">Get trace files<a class="anchor" aria-label="anchor" href="#get-trace-files"></a>
</h3>
<p>This function downloads files to your machine - it does not load them into your R session - but prints out where the files are for your information.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">traces1</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/bold_trace.html">bold_trace</a></span><span class="op">(</span>ids <span class="op">=</span> <span class="st">'ACRJP618-11'</span>, progress <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">traces1</span></span>
<span><span class="co">#&gt;&lt;bold trace files&gt; </span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt;./bold_trace_files/ACRJP618-11[LepF1,LepR1]_F.ab1</span></span>
<span><span class="va">traces2</span> <span class="op">&lt;-</span> <span class="fu">bold</span><span class="fu">::</span><span class="fu"><a href="reference/bold_trace.html">bold_trace</a></span><span class="op">(</span>ids <span class="op">=</span> <span class="st">'SALIX204-08'</span>, progress <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">traces2</span></span>
<span><span class="co">#&gt; &lt;bold trace files&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ./bold_trace_files/ACRJP618-11[LepF1,LepR1]_F.ab1</span></span>
<span><span class="co">#&gt; ./bold_trace_files/SALIX204-08[MatK-1RKIM-f,MatK-3FKIM-r]_F.ab1</span></span>
<span><span class="co">#&gt; ./bold_trace_files/SALIX204-08[MatK-1RKIM-f,MatK-3FKIM-r]_R.ab1</span></span>
<span><span class="co">#&gt; ./bold_trace_files/SALIX204-08[rbcLa-F,rbcLa-R]_F.ab1</span></span>
<span><span class="co">#&gt; ./bold_trace_files/SALIX204-08[rbcLa-F,rbcLa-R]_R.ab1</span></span></code></pre></div>
<p>Then to read the files, you can use <code><a href="reference/bold_trace.html">bold::bold_read_trace()</a></code> with either the boldtrace object or a subset of the paths.</p>
<pre class="{r}"><code>bold_read_trace(trace1)
#&gt; $`ACRJP618-11[LepF1,LepR1]_F.ab1`
#&gt; Number of datapoints: 8877
#&gt; Number of basecalls: 685
#&gt; 
#&gt; Primary Basecalls: NNNNNNNNNNNNNNNNNNGNNNTTGAGCAGGNATAGTAGGANCTTCTCTTAGTCTTATTATTCGAACAGAATTAGGAAATCCAGGATTTTTAATTGGAGATGATCAAATCTACAATACTATTGTTACGGCTCATGCTTTTATTATAATTTTTTTTATAGTTATACCTATTATAATTGGAGGATTTGGTAATTGATTAGTTCCCCTTATACTAGGAGCCCCAGATATAGCTTTCCCTCGAATAAACAATATAAGTTTTTGGCTTCTTCCCCCTTCACTATTACTTTTAATTTCCAGAAGAATTGTTGAAAATGGAGCTGGAACTGGATGAACAGTTTATCCCCCACTGTCATCTAATATTGCCCATAGAGGTACATCAGTAGATTTAGCTATTTTTTCTTTACATTTAGCAGGTATTTCCTCTATTTTAGGAGCGATTAATTTTATTACTACAATTATTAATATACGAATTAACAGTATAAATTATGATCAAATACCACTATTTGTGTGATCAGTAGGAATTACTGCTTTACTCTTATTACTTTCTCTTCCAGTATTAGCAGGTGCTATCACTATATTATTAACGGATCGAAATTTAAATACATCATTTTTTGATCCTGCAGGAGGAGGAGATCCAATTTTATATCAACATTTATTTTGATTTTTTGGACNTCNNNNAAGTTTAAN
#&gt; 
#&gt; Secondary Basecalls:

bold_read_trace(trace2$ab1[1:2])
&gt;# $`ACRJP618-11[LepF1,LepR1]_F.ab1`
&gt;# Number of datapoints: 8877
&gt;# Number of basecalls: 685
&gt;# 
&gt;# Primary Basecalls: NNNNNNNNNNNNNNNNNNGNNNTTGAGCAGGNATAGTAGGANCTTCTCTTAGTCTTATTATTCGAACAGAATTAGGAAATCCAGGATTTTTAATTGGAGATGATCAAATCTACAATACTATTGTTACGGCTCATGCTTTTATTATAATTTTTTTTATAGTTATACCTATTATAATTGGAGGATTTGGTAATTGATTAGTTCCCCTTATACTAGGAGCCCCAGATATAGCTTTCCCTCGAATAAACAATATAAGTTTTTGGCTTCTTCCCCCTTCACTATTACTTTTAATTTCCAGAAGAATTGTTGAAAATGGAGCTGGAACTGGATGAACAGTTTATCCCCCACTGTCATCTAATATTGCCCATAGAGGTACATCAGTAGATTTAGCTATTTTTTCTTTACATTTAGCAGGTATTTCCTCTATTTTAGGAGCGATTAATTTTATTACTACAATTATTAATATACGAATTAACAGTATAAATTATGATCAAATACCACTATTTGTGTGATCAGTAGGAATTACTGCTTTACTCTTATTACTTTCTCTTCCAGTATTAGCAGGTGCTATCACTATATTATTAACGGATCGAAATTTAAATACATCATTTTTTGATCCTGCAGGAGGAGGAGATCCAATTTTATATCAACATTTATTTTGATTTTTTGGACNTCNNNNAAGTTTAAN
&gt;# 
&gt;# Secondary Basecalls: 
&gt;# 
&gt;# $`SALIX204-08[MatK-1RKIM-f,MatK-3FKIM-r]_F.ab1`
&gt;# Number of datapoints: 16299
&gt;# Number of basecalls: 877
&gt;# 
&gt;# Primary Basecalls: NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTACGACTCTTTCTTCACGAGTATTGGAATTTGAACAGTCGTATTATTCCAAAGAAATCTATTTCTTTTTTTGCAAAAAAGACTCCAAGATTCTTCTTGTTCTTATATAATTCTCATGTATATGAATACGAGTCCGTTTTCTTTTTTCTTTGTAATCAATCCTTTCATTTCCGATTAACATTTTCTCAGGTCTTTCTTGAGCGAATATATTTCTATGGAAAAATAGAACATTTTGTAGAAGTCTTTACTAAGGATTGGGGGGACAGCCTATGCTTGCTCAAGGATCCTTTCATACATTATCTTAGATATCAAGGAAAATCCATTTTTGTCTCAAAGGATACGCCTCTTCTGATGAAAAAATGGAAATATTACCTTGTCAATTTATGTCAATGTCATTTTGATGTGTGCTTTCAACCCCCCAGGATCCATATAAACCCATTTTCATTATACAAGCATTCNTTCGCCTTATTAGGTTATCTTTCNAGTTCNAGTGNNCNACTNANCCTTTCAGTGGTACGGAGTCAAATGCTANAAAATGCATTTCTAATANATAATATTATGAATAAACTCGATACAACAGTTTCAATTANTCCTTTGATTGGATCATTAGCANAACTGAAATTTTGTNACGCNGTANGACATCCCATTAGTNAAACNGGCCTGNGCCGATTTTTCGGATTCTGATATTANCNACNGANTTGTCCNTNNNNNCAGAAATCNNNNTCATTNTNANNNCNGATCCNNNAGAAANANGAANTNNTATCNAATANAATATANNNTTCNNCTTTCNTGNGNNANNNCTTNGGCTCNTNANCNNNNANTNNNNAANNGN
&gt;# 
&gt;# Secondary Basecalls: </code></pre>
</div>
<div class="section level3">
<h3 id="large-data">Large data<a class="anchor" aria-label="anchor" href="#large-data"></a>
</h3>
<p>Sometimes with <code><a href="reference/bold_seq.html">bold_seq()</a></code>, <code><a href="reference/bold_seqspec.html">bold_seqspec()</a></code> and <code>bold_specimen()</code> you request a lot of data, which can cause problems due to BOLD’s servers.</p>
<p>An example is the taxonomic name <em>Arthropoda</em>. When you send a request like <code>bold_seq(taxon = "Arthropoda")</code>, <code>bold_seqspec(taxon = "Arthropoda")</code> or <code>bold_specimen(taxon = "Arthropoda")</code> BOLD attempts to give you back sequences and/or specimen information for all records under <em>Arthropoda</em>. This, as you can imagine, is a lot of sequences.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://docs.ropensci.org/taxize/" class="external-link">"taxize"</a></span><span class="op">)</span></span></code></pre></div>
<p>Using <code>taxize::downstream</code> get children of <em>Arthropoda</em></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">downstream</span><span class="op">(</span><span class="st">"Arthropoda"</span>, db <span class="op">=</span> <span class="st">"ncbi"</span>, downto <span class="op">=</span> <span class="st">"class"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  1 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  Arthropoda</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 1 </span></span>
<span><span class="co">#&gt; ● Found: 1 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="va">nms</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">Arthropoda</span><span class="op">$</span><span class="va">childtaxa_name</span></span></code></pre></div>
<p>Optionally, check that the name exists in BOLD’s data. Any that are not in BOLD will give back a row of NAs</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">checks</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/bold_tax_name.html">bold_tax_name</a></span><span class="op">(</span><span class="va">nms</span><span class="op">)</span></span>
<span><span class="co"># all is good</span></span>
<span><span class="va">checks</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;     taxid         taxon tax_rank tax_division parentid</span></span>
<span><span class="co">#&gt; 1   26059   Pycnogonida    class     Animalia       20</span></span>
<span><span class="co">#&gt; 2      63     Arachnida    class     Animalia       20</span></span>
<span><span class="co">#&gt; 3      74   Merostomata    class     Animalia       20</span></span>
<span><span class="co">#&gt; 4  493944     Pauropoda    class     Animalia       20</span></span>
<span><span class="co">#&gt; 5   80390      Symphyla    class     Animalia       20</span></span>
<span><span class="co">#&gt; 6      85     Diplopoda    class     Animalia       20</span></span>
<span><span class="co">#&gt; 7      75     Chilopoda    class     Animalia       20</span></span>
<span><span class="co">#&gt; 8      82       Insecta    class     Animalia       20</span></span>
<span><span class="co">#&gt; 9     372    Collembola    class     Animalia       20</span></span>
<span><span class="co">#&gt; 10 734357       Protura    class     Animalia       20</span></span>
<span><span class="co">#&gt; 11     84     Remipedia    class     Animalia       20</span></span>
<span><span class="co">#&gt; 12     73 Cephalocarida    class     Animalia       20</span></span>
<span><span class="co">#&gt; 13     68  Branchiopoda    class     Animalia       20</span></span>
<span><span class="co">#&gt; 14 765970   Hexanauplia    class     Animalia       20</span></span>
<span><span class="co">#&gt; 15     69  Malacostraca    class     Animalia       20</span></span>
<span><span class="co">#&gt; 16 889450 Ichthyostraca    class     Animalia       20</span></span>
<span><span class="co">#&gt; 17     NA          &lt;NA&gt;     &lt;NA&gt;         &lt;NA&gt;       NA</span></span>
<span><span class="co">#&gt; 18     80     Ostracoda    class     Animalia       20</span></span></code></pre></div>
<p>Then pass those names to <code><a href="reference/bold_seq.html">bold_seq()</a></code>. You could pass all names in at once, but we’re trying to avoid the large data request problem here, so run each one separately with <code>lapply</code> or a for loop like request.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">nms</span>, <span class="va">bold_seq</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Get citation information for <code>bold</code> in R by running: <code>citation(package = 'bold')</code></p>
</div>
<div class="section level2">
<h2 id="meta">Meta<a class="anchor" aria-label="anchor" href="#meta"></a>
</h2>
<ul>
<li>Please <a href="https://github.com/ropensci/bold/issues" class="external-link">report any issues or bugs</a>
</li>
<li>License: MIT</li>
<li>Get citation information for <code>bold</code> in R doing <code>citation(package = 'bold')</code>
</li>
<li>Please note that this project is released with a <a href="https://ropensci.org/code-of-conduct/" class="external-link">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=bold" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/ropensci/bold/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ropensci/bold/issues" class="external-link">Report a bug</a></li>
</ul>
</div>



<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing bold</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Salix Dubois <br><small class="roles"> Author, maintainer </small>  </li>
<li>
<a href="https://scottchamberlain.info/" class="external-link">Scott Chamberlain</a> <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-1444-9135" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="r-universe">
<h2 data-toc-skip>R-universe</h2>
<ul class="list-unstyled">
<li><p><a href="https://ropensci.r-universe.dev/bold" class="external-link"><img src="https://ropensci.r-universe.dev/badges/bold" alt="bold status badge"></a></p></li>
</ul>
</div>

  </aside>
</div>


    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
